WIP global rephrasing (#222)
meilisearch/documentation: guides/main_concepts/documents.md

# Documents

**Documents** are objects composed of fields which can store any type of data.</br>

> Each **field** contains an **attribute** and its associated data.

Reviewer: Since I added value on the `schema`.
```suggestion
Each **field** contains an **attribute** and its associated value.
```

![document structure](/document_structure.svg)

#### Wording

- **Attribute**: An attribute is like a key (`"id"`, `"title"`, `"description"` and `"type"`).
- **Field**: A field, or a key-value pair, is a set of two data items linked together. Here, an attribute and its corresponding data (i.e, `"title": "Interstellar"`).
- # **Document**: A document is an object which contains a list of fields in curly brackets.
- **Attribute**: An attribute is the name of a field, like a key.
- **[Field](/guides/main_concepts/documents.md#fields)**: A field, or a key-value pair, is a set of two data items linked together: an attribute and its corresponding data
- **Document**: A document is an object which collects a list of fields between curly brackets.
- **[Primary key](/guides/main_concepts/documents.md#primary-key)**: The attribute of the field which contains the unique identifier of the documents.
- **[Document id](/guides/main_concepts/documents.md#document-id)**: The unique value of a document.

>   > > > > > > > cf20813fb99994943b619a3e29426e08e8bd2c71

Reviewer: Merge conflict   ##error.
```suggestion
```

Author: Thanks a lot!

## Structure

> Documents are stored in `JSON format`.<br/>

Reviewer: This changes the sentence to an incorrect information. 
Documents are sent to MeiliSearch using the JSON format. Once it is in meilisearch it is not stored in JSON format.
```suggestion
Documents are sent to MeiliSearch in `JSON format`.<br/>
```

Author: Thanks for the correction!

> When using the [route to add new documents](/references/documents.md#add-or-update-documents), all documents must be sent in an array even if there is only one document.

Reviewer: ```suggestion
When using the [route to add new documents](/references/documents.md#add-or-update-documents), all documents must be sent in an array **even if there is only one document**.
```

]'
```

In order to be indexed, each **document must contain** [the primary key field](/guides/main_concepts/documents.md#primary-key).

## Fields

> By default, all fields included in a document are <clientGlossary word="searchable" /> and automatically <clientGlossary word="displayed" />.

Reviewer: I'm not sure why only `displayed` is automatic.
```suggestion
By default, all fields included in a document are <clientGlossary word="searchable" /> and <clientGlossary word="displayed" />.
```

> You can adjust how a field get handled by MeiliSearch in the settings <Badge text="soon" type="warn"/>. It can either be searchable or displayed, or none of both. In the latter case, the field will be completely ignored when a document is sent.

Reviewer: ```suggestion
You can adjust how a field get handled by MeiliSearch in the settings <Badge text="soon" type="warn"/>. It can either be searchable or displayed, both, or none of both. In the latter case, the field will be completely ignored when a document is sent.
```

]
```

### Setting the primary key

> MeiliSearch can deduce which field is the `primary key` in several ways.

Reviewer: Is it deducing when you give the information using methods 2 and 3?

Author: Yes, you're right! What about:
```suggestion
There are several ways for MeiliSearch to know which field is the `primary key` in several ways.
```

Reviewer: yes! Thanks!

- MeiliSearch [automatically infers the primary key](/guides/main_concepts/documents.md#meilisearch-infers-your-primary-key) based on your first document.
- Set it [on index creation](/references/indexes.md#create-an-index)
- Set it [on document addition](/references/documents.md#add-or-replace-documents)

#### MeiliSearch infers your primary key

> If no primary key has been given either at index creation time or through the addition of document, the primary key will be searched in the first document sent.

Reviewer: ```suggestion
If no primary key has been given either when the index was created or as a parameter of the add documents route, the primary key will be searched in the first document sent.
```

Author: Much better!

MeiliSearch will search for an attribute that contains the string `id` in a case-insensitive manner (i.e, `uid`, `MovieId`, `ID`, `123id123`).<br>

> If none has been found, no documents will be added.

Reviewer: ```suggestion
If no corresponding attribute has been found, the index will have no known primary key, and therefore, no documents will be added.
```

Author: Much better too! Thanks!

"id": "@BI+* ^5h2%"
```

> The document addition request in MeiliSearch is <!-- prettier-ignore -->[atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)). Thus, for example, if you submit 200 documents in one go and it happens one of them contains a wrongly formatted primary key, an error will occur and result in no additions.

Reviewer: ```suggestion
The document addition request in MeiliSearch is <!-- prettier-ignore -->[atomic](https://en.wikipedia.org/wiki/Atomicity_(database_systems)). Thus, for example, if you submit 200 documents in one go and it happens one of them contains a wrongly formatted document id, an error will occur and result in no additions.
```
